src/wasm-interpreter.h

```diff
@@ -4300,11 +4607,105 @@ class ModuleRunnerBase : public ExpressionRunner<SubType> {
+      return Flow();
+    }
+
+    // We were not resuming, so this is a new suspend that we must execute.
+    Literals arguments;
+    Flow flow = self()->generateArguments(curr->operands, arguments);
+    if (flow.breaking()) {
+      return flow;
+    }
 
-  Flow visitSuspend(Suspend* curr) { return Flow(NONCONSTANT_FLOW); }
-  Flow visitResume(Resume* curr) { return Flow(NONCONSTANT_FLOW); }
+    if (!self()->currContinuation) {
+      trap("no continuation to suspend");
+    }
+    // Copy the continuation (the old one cannot be resumed again).
+    auto old = self()->currContinuation;
+    assert(old->executed);
+    auto new_ = std::make_shared<ContData>(old->func, old->type);
+    // Switch to the new continuation, so that as we unwind, we will save the
+    // information we need to resume it later in the proper place.
+    self()->currContinuation = new_;
+    // We will resume from this precise spot, when the new continuation is
+    // resumed.
+    new_->resumeExpr = curr;
+    return Flow(SUSPEND_FLOW, curr->tag, std::move(arguments));
+  }
+  Flow visitResume(Resume* curr) {
+    Literals arguments;
+    Flow flow = self()->generateArguments(curr->operands, arguments);
+    if (flow.breaking()) {
+      return flow;
+    }
+    flow = self()->visit(curr->cont);
+    if (flow.breaking()) {
+      return flow;
+    }
+
+    // Get and execute the continuation.
+    auto contData = flow.getSingleValue().getContData();
+    if (contData->executed) {
+      trap("continuation already executed");
+    }
+    contData->executed = true;
+    Name func = contData->func;
+    self()->currContinuation = contData;
+    if (contData->resumeExpr) {
+      // There is an expression to resume execution at, so this is not the first
+      // time we run this function. Mark us as resuming, until we reach that
+      // expression.
+      self()->resuming = true;
+    }
+#if WASM_INTERPRETER_DEBUG
+    std::cout << self()->indent() << "resuming func " << func << '\n';
+#endif
+    Flow ret = callFunction(func, arguments);
+#if WASM_INTERPRETER_DEBUG
+    std::cout << self()->indent() << "finished resuming, with " << ret << '\n';
+#endif
+    if (ret.suspendTag) {
+      // See if a suspension arrived that we support.
+      for (size_t i = 0; i < curr->handlerTags.size(); i++) {
+        auto handlerTag = curr->handlerTags[i];
+        if (handlerTag == ret.suspendTag) {
+          // Switch the flow from suspending to branching.
+          ret.suspendTag = Name();
+          ret.breakTo = curr->handlerBlocks[i];
+          // Add the continuation as the final value being sent.
+          ret.values.push_back(Literal(self()->currContinuation));
+          // We are not longer processing that continuation.
+          self()->currContinuation.reset();
+          return ret;
+        }
+      }
+      // No handler worked out, keep propagating.
+      return ret;
+    }
+    // No suspension; all done.
+    return Flow();
```

Comment:
We still need to return the value the continuation returned.
```suggestion
    return ret;
```